src/thread-utils.h

```diff
@@ -7,28 +7,38 @@
 #ifndef INCLUDE_thread_utils_h__
 #define INCLUDE_thread_utils_h__
 
-#if defined(__GNUC__) && defined(GIT_THREADS)
-# if (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 1))
-#  error Atomic primitives do not exist on this version of gcc; configure libgit2 with -DTHREADSAFE=OFF
+#if defined(GIT_THREADS)
+# if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L && !defined(__STDC_NO_ATOMICS__)
+#  define __GIT_THREADS_USE_C11 1
```

Comment:
I'm afraid that using C11 for atomics won't work for us. Our project is trying to conform to C90, so we simply cannot use C11 features at all. I agree that it kind of sucks that we need to rely on GCC builtins for our atomic support instead, but for C90 there is no standardized way to do that.

Reply:
> Our project is trying to conform to C90

I see it now in the readme, but this requirement is ill-documented. CMake makes no attempt to configure the compiler that way, so certainly most of distributions you package for have a separate default. I see a fair amount of code `#if` in the repo around C99 compilers, so I don't see why properly-checked progressive enhancement is a problem.

> … but for C90 there is no standardized way to do that.

It's kind of absurd to say this when 2 lines below there's a bespoke variant for Windows. The GCC builtins are known to be deficient; that's why they went through standardization. If I use `#if defined(__clang__)` and its builtins, is that somehow better? C11 atomics are also the only guaranteed-correct way for platforms now; if I write `#if defined(__APPLE__) || defined(__ANDROID__) || defined(__Fuchsia__)`, is that somehow better? If so, why?